### PR TITLE
[TLS/coverage] Migrate to UTS + Bug

### PR DESCRIPTION
**Pull request by gpotter2**
This PR : 
- migrates the tests from `run_tests_tls_netaccess.sh` to a python form using .uts file
- tries to fix coverage issues https://coverage.readthedocs.io/en/coverage-4.3.4/cmd.html#execution

It also fixes https://github.com/secdev/scapy/issues/505, to make the tests work.

@mtury Can you help me with openssl ? As you can see on the travis logs of this PR, i tried passing it the ca_cert.pem without any success...

### CODE REVIEW

File: scapy/autorun.py [hunk 1]
```diff
@@ -78,6 +78,8 @@ def __init__(self):
             self.s = ""
         def write(self, x):
             self.s += x
+        def flush(self):
+            pass
```
Comment (gpotter2) [inline scapy/autorun.py]: This is required, otherwise multiprocessing will (for some reason) crash

File: test/tls/travis_test_client.py [hunk 2]
```diff
@@ -25,18 +24,29 @@
...
+    t = TLSClientAutomaton(client_hello=ch, data=send_data)
+    t.run()
+
+from travis_test_server import run_tls_test_server
+
+def test_tls_client(suite, version, q):
+    msg = "TestC_" + suite + "_data"
+    # Run server
+    q_ = multiprocessing.Manager().Queue()
+    th_ = multiprocessing.Process(target=run_tls_test_server, args=(msg, q_))
```
Comment (gpotter2) [inline test/tls/travis_test_client.py]: Using a multiprocessing and it's wrapper allow us to kill the automaton when timed out

**Comment by mtury** [comment]
Using `-CAfile` is a fine idea!

As far as I understand, `verify return:1` actually shows that openssl was able to verify the server certificate. I have no idea why openssl would send this to stderr (something is sent to stderr anyway, even if the `-CAfile` option is removed). Maybe these lines could be filtered out when checking the err return of communicate? It would be dirty, but then again, we're trying to do openssl scripting...

File: scapy/layers/tls/automaton.py [hunk 3]
```diff
@@ -735,7 +735,7 @@ def BIND_AND_WAIT(self):
                                                                 self.local_port)
             return
         self.socket, addr = s.accept()
-        self.remote_ip, self.remote_port = addr
+        self.remote_ip, self.remote_port = addr, self.local_port
```
Comment (mtury) [inline scapy/layers/tls/automaton.py]: It seems this is not the way to fix issue #505.

Comment (gpotter2) [inline scapy/layers/tls/automaton.py]: You're right

**Comment by gpotter2** [comment]
@mtury That is a way better, what do you think ?
https://docs.python.org/2/library/socket.html#socket.socket.getpeername

**Comment by gpotter2** [comment]
https://github.com/secdev/scapy/pull/496 and this PR combined, the coverage should really improve

**Comment by guedou** [comment]
Could you remove  `test/tls/run_tests_tls_netaccess.old` ?

**Comment by mtury** [comment]
Also, it seems that the `import threading` in both `travis_test_client.py` and `travis_test_server.py` could be removed.

**Comment by gpotter2** [comment]
@mtury @guedou That should do it.
Appveyor failed because of networking

**Comment by p-l-** [comment]
@gpotter2: Appveyor restarted and OK, waiting for Travis / MacOS...

**Comment by p-l-** [comment]
@gpotter2 your comment should be in the code, rather than on Github (where they will be lost for the next person who will read your code). Can you submit another PR with just that? (I didn't want to block the merge for that.)

**Pull request merged**